homeassistant/components/yamaha/__init__.py

```diff
@@ -1 +1,72 @@
 """The yamaha component."""
+
+from __future__ import annotations
+
+from functools import partial
+import logging
+
+import rxv
+
+from homeassistant.components import ssdp
+from homeassistant.config_entries import ConfigEntry
+from homeassistant.const import CONF_HOST, Platform
+from homeassistant.core import HomeAssistant
+
+from .const import CONF_SERIAL, CONF_UPNP_DESC, DOMAIN
+from .yamaha_config_info import YamahaConfigInfo
+
+PLATFORMS = [Platform.MEDIA_PLAYER]
+
+_LOGGER = logging.getLogger(__name__)
+
+
+async def get_upnp_desc(hass: HomeAssistant, host: str) -> str:
+    """Get the upnp description URL for a given host, using the SSPD scanner."""
+    ssdp_entries = await ssdp.async_get_discovery_info_by_st(hass, "upnp:rootdevice")
+    matches = [w for w in ssdp_entries if w.ssdp_headers.get("_host", "") == host]
+    upnp_desc = None
+    for match in matches:
+        if upnp_desc := match.ssdp_location:
+            break
+
+    if not upnp_desc:
+        _LOGGER.warning(
+            "The upnp_description was not found automatically, setting a default one"
+        )
+        upnp_desc = f"http://{host}:49154/MediaRenderer/desc.xml"
+    return upnp_desc
+
+
+async def async_setup_entry(hass: HomeAssistant, entry: ConfigEntry) -> bool:
+    """Set up Yamaha from a config entry."""
+    if entry.data.get(CONF_UPNP_DESC) is None:
+        hass.config_entries.async_update_entry(
```

Comment:
How does this work?

Like I am known with the technique to do this in the config flow discovery step, so for example when someone setup the integration using the host and then we find either a zeroconf or DHCP flow that we can link to it (for example if the entry uses the same host and is loaded) and then we update the entry to have that unique identifier. Examples of this are `fritzbox` and `reolink`, both do this in the DHCP step I believe.

From my understanding this does the same, but I am wondering how the order of discovery works, because if the integration is setup and after that the SSDP picks up the advertisement, this code is not run anymore.

So I am not really sure what is best here, but I do want to share this info with you to at stir up the discussion.